src/Platform_Core/org/lobobrowser/context/AlgorithmDigest.java

```
@@ -0,0 +1,131 @@
+      return null;
+
+    return strongestAlgDigests(hashes);
+
+  }
+
+  private static List<AlgorithmDigest> getHashes(final String[] tokens) {
+    final List<AlgorithmDigest> hashes = new ArrayList<AlgorithmDigest>();
+
+    for (int i = 0; i < tokens.length; i++) {
+      // TODO: check token syntax against
+      // https://www.w3.org/TR/SRI/#the-integrity-attribute
+      final int hyphen = tokens[i].indexOf("-");
+      if (hyphen < 0) {
+        continue;
+      }
+
+      final String alg = tokens[i].substring(0, hyphen);
+      if (str.containsKey(alg)) {
+        final AlgorithmDigest ag = new AlgorithmDigest(alg, tokens[i].substring(hyphen + 1), str.get(alg));
+        hashes.add(ag);
+
+      }
+
+    }
+    return hashes;
+  }
+
+  public static List<AlgorithmDigest> strongestAlgDigests(final List<AlgorithmDigest> hashes) {
+    Collections.sort(hashes, (a, b) -> a.compareTo(b));
+
+    System.out.println("Hashes: " + Arrays.toString(hashes.toArray()));
+    final Integer strongest = hashes.get(0).strength;
+
+    final List<AlgorithmDigest> result = hashes.stream().filter((h) -> h.strength == strongest)
+        .collect(Collectors.toList());
+
+    System.out.println("result: " + Arrays.toString(result.toArray()));
+    return result;
+  }
+
+  public String getHash(final byte[] input) {
+    final String alg = algorithm.substring(0, 3) + "-" + algorithm.substring(3);
+    System.out.println("Algorithm used is: " + alg);
+    try {
+      final MessageDigest md = MessageDigest.getInstance(alg); // returns an object that implements alg
+      md.update(input); // updates digest using input array of bytes
+      final byte[] digestedBytes = md.digest(); // completes hash, returns array of bytes
+      return Base64.getEncoder().encodeToString(digestedBytes); // converts array of bytes into base64-encoded string
+    } catch (final Exception e) {
+      System.out.println("Exception: " + e.getMessage());
+    }
+    return "";
```

I think it is better to return `null` on failure, instead of empty string.